src/components/structures/RoomView.js

```diff
@@ -191,18 +191,32 @@ module.exports = React.createClass({
             newState.room = MatrixClientPeg.get().getRoom(newState.roomId);
         }
 
+        if (this.state.roomId !== newState.roomId) {
```

Comment:
this is somewhat in conflict with the comment at line 188: "This does assume that the roomID will not change for the lifetime of the RoomView instance". If we're changing room, LoggedInView gives us a new RoomView.

I think that this is hit when we resolve a room id after switching to room by alias, though, so it's sort-of valid, though I'd prefer it to be expressed more like `if (this.state.roomId === null && newState.roomId !== null)`.

Reply:
yes, I think so - removed the scroll state saving here too as it'll now never be hit